extensions/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsLoadSpec.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.storage.hdfs;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Preconditions;
+import com.metamx.common.ISE;
+import io.druid.segment.loading.DataSegmentPuller;
+import io.druid.segment.loading.LoadSpec;
+import io.druid.segment.loading.SegmentLoadingException;
+import org.apache.hadoop.fs.Path;
+
+import java.io.File;
+import java.util.Map;
+
+/**
+ *
+ */
+@JsonTypeName(HdfsStorageDruidModule.SCHEME)
+public class HdfsLoadSpec implements LoadSpec
+{
+  private final Path path;
+  final HdfsDataSegmentPuller puller;
+  @JsonCreator
+  public HdfsLoadSpec(
+      @JacksonInject Map<String, DataSegmentPuller> pullers,
+      @JsonProperty(value = "path", required = true) String path
+  ){
+    Preconditions.checkNotNull(path);
+    this.path = new Path(path);
+    final DataSegmentPuller puller = pullers.get("hdfs");
+    if(!(puller instanceof HdfsDataSegmentPuller)){
+      throw new ISE("hdfs not tied to HdfsDataSegmentPuller");
+    }
+    this.puller = (HdfsDataSegmentPuller)puller;
+  }
+  @JsonProperty("path")
+  public final String getPathString(){
+    return path.toString();
+  }
+
+  @Override
+  public Long putSegment(File outDir) throws SegmentLoadingException
```

Comment:
DataSegment exists, that has similar functionality to what you describe, but this is a servant to DataSegment.

Originally this effectively returned Void. This was changed to Long to allow some level of sanity checking in the future. I'm trying to think of things that would be helpful for DataSegment (or its handler) to know that isn't already in DataSegment. The only thing I could think of would be some sort of handler to the files that were loaded (so like a thin wrapper for `Iterable<File>` or something) then whatever is handling the DataSegment could use that info in whatever way it needs.

The LoadSpec pretty much only knows "where do I get files" and "how do I tell the puller the correct place to get them and the correct place to put them", so its knowledge realm is pretty thin.